iceoryx_posh/include/iceoryx_posh/popo/modern_api/typed_publisher.hpp

```diff
@@ -27,6 +27,7 @@ template <typename T, typename base_publisher_t = BasePublisher<T>>
 class TypedPublisher : public base_publisher_t
 {
     static_assert(!std::is_void<T>::value, "Type must not be void. Use the UntypedPublisher for void types.");
+    static_assert(std::is_default_constructible<T>::value, "The TypedPublisher requires default-constructable types.");
```

Comment:
Why does this have to be the case? Are we not constructing the data into raw memory, given arbitrary constructor arguments? It may be a limitation right now, but I imagine this due to the problem itself, only due to the current design. I may be wrong though, there may be a deeper underlying reason.

Reply:
The answer lies here: https://github.com/eclipse/iceoryx/pull/335#discussion_r518087695

In short, apparently the automotive world expects samples to be default-constructable and not doing so will be confusing.